src/views/Characters/CharacterList.test.jsx

```diff
@@ -0,0 +1,114 @@
+import { render, screen } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import App from '../../App';
+import { setupServer } from 'msw/node';
+import { rest } from 'msw';
+
+const server = setupServer(
+  rest.get('https://rickandmortyapi.com/api/character', (req, res, ctx) => {
+    return res(
+      ctx.json({
```

Comment:
for the mocked returned json, you can slim the data down to clean up your test a bit (i.e. only return data in the most basic shape you need by deleting the episode array, location, etc.)